dbt-athena/src/dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -13,17 +13,14 @@
   {% set temp_schema = config.get('temp_schema') %}
   {% set target_relation = this.incorporate(type='table') %}
   {% set existing_relation = load_relation(this) %}
-  -- If using insert_overwrite on Hive table, allow to set a unique tmp table suffix
-  {% if unique_tmp_table_suffix == True and strategy == 'insert_overwrite' and table_type == 'hive' %}
-    {% set tmp_table_suffix = adapter.generate_unique_temporary_table_suffix() %}
+  -- Generate a unique tmp table suffix if required
+  {% if unique_tmp_table_suffix == True %}
+    {% set raw_suffix = adapter.generate_unique_temporary_table_suffix() %}
+    {% set tmp_table_suffix = raw_suffix.replace('-', '_') %}
```

Comment:
please move this inside the generate_unique_temporary_table_suffix.
replacing `-` with `_` make sense also for an athena/trino SQL prospective. 

Also, remember to adapt the functional tests, otherwise they will fail.

Reply:
#754 was merged, therefore please remove this, as not needed anymore